ocaml/typing/typemod.ml

```diff
@@ -802,7 +802,7 @@ let merge_constraint initial_env loc sg lid constr =
         let mty = Mtype.scrape_for_type_of ~remove_aliases sig_env mty in
         let mty =
           remove_modality_and_zero_alloc_variables_mty sig_env
-            ~zap_modality:Mode.Modality.Value.zap_to_floor mty
+            ~zap_modality:Mode.Modality.Value.zap_to_id mty
```

Comment:
I think maybe this is fine for now, but really the problem is that `with module` shouldn't touch `val`s at all. The manual describes `with module` as affecting only type equalities. If that were the case, this would all be moot.

Or maybe we want `with module` to also affect modalities on `val`s? That actually seems like perhaps the right idea. Is this documented anywhere?

Reply:
`with modules` was overlooked when we designed `val_modalities`, for the reason in your first paragraph.

But now i think about this, since `with module` is strenghtening, then it might make sense that `with module` strenghthens the modalities. I will look.